lib/Parser.js

```diff
@@ -1976,18 +1978,46 @@ class Parser extends Tapable {
 
 	getCommentOptions(range) {
 		const comments = this.getComments(range);
-		if (comments.length === 0) return null;
-		const options = comments.map(comment => {
-			try {
-				let val = vm.runInNewContext(
-					`(function(){return {${comment.value}};})()`
-				);
-				return val;
-			} catch (e) {
-				return {};
-			}
-		});
-		return options.reduce((o, i) => Object.assign(o, i), {});
+		const { current } = this.state;
+		let options = null;
+		const errors = [];
+		if (comments.length !== 0) {
+			const optionsList = comments.map(comment => {
+				const context = {};
+				if (this.hooks.commentOptions.call(comment, context) !== undefined) {
+					return {};
+				}
+				try {
+					let val = vm.runInNewContext(
+						`(function(){return {${comment.value}};})()`,
+						context
+					);
+					return val;
+				} catch (e) {
+					if (comment.loc) {
+						e.message += ` (${comment.loc.start.line}:${
+							comment.loc.start.column
+						})`;
+					}
+					errors.push(e);
+					return {};
+				}
+			});
+			options = optionsList.reduce((o, i) => Object.assign(o, i), {});
```

Comment:
We can simplify merging options and avoid `reduce` after `map`.
It's enough to just use `forEach` and `Object.assign` inside.

From:
```js
const optionsList = comments.map(comment => {
    const context = {};
    if (this.hooks.commentOptions.call(comment, context) !== undefined) {
        return {};
    }
    try {
        let val = vm.runInNewContext(
            `(function(){return {${comment.value}};})()`,
            context
        );
        return val;
    } catch (e) {
        if (comment.loc) {
            e.message += ` (${comment.loc.start.line}:${
                comment.loc.start.column
            })`;
        }
        errors.push(e);
        return {};
    }
});
options = optionsList.reduce((o, i) => Object.assign(o, i), {});
```
To:
```js
options = {}
comments.forEach(comment => {
    const context = {};
    if (this.hooks.commentOptions.call(comment, context) !== undefined) {
        return;
    }
    try {
        let val = vm.runInNewContext(
            `(function(){return {${comment.value}};})()`,
            context
        );
        Object.assign(options, val);
    } catch (e) {
        if (comment.loc) {
            e.message += ` (${comment.loc.start.line}:${
                comment.loc.start.column
            })`;
        }
        errors.push(e);
    }
});
```